### ISSUE TITLE
Rule request: informative-docs

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [Microsoft/tslint-microsoft-contrib#457]
#### TypeScript code being linted

```typescript
/**
 * Request metadata.
 */
interface IRequestMetadata {
    /**
     * The user id
     */
    userId: string;
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "completed-docs": true
    }
}
```

#### Actual behavior

Many teams require all things be documented, such as with TSLint core's [`completed-docs`](https://palantir.github.io/tslint/rules/completed-docs/). The idea of requiring code be well-documented is nice, but you end up with the above useless comments. 

#### Expected behavior

It'd be nice to have a rule that requires the documentation for something not be equivalent to just the thing's name. If we strip out non-alphabet characters, common words such as "the" or "a", and lowercase everything, they shouldn't be the same.

Posting here instead of TSLint core, as I expect this is too opinionated for the core ruleset.

**Comment by sahalsaad** [comment]
Hi, I would like to work on this if no one currently working on it. Thanks

**Comment by JoshuaKGoldberg** [comment]
I have a working version of this internally and will post it here soon.
